packages/data-schema/src/SchemaProcessor.ts

```diff
@@ -911,17 +912,20 @@ function mapToNativeAppSyncAuthDirectives(
       };
       rule.groups.forEach((group) => groupProvider.get(provider)?.add(group));
     } else {
+      //generalProviderUsed.add(provider);
       rules.add(provider);
     }
   }
 
   groupProvider.forEach((groups, provider) => {
-    rules.add(
-      `${provider}(cognito_groups: [${Array.from(groups)
-        .map((group) => `"${group}"`)
-        .join(', ')}])`,
-    );
-    // example: (cognito_groups: ["Bloggers", "Readers"])
+    if(!generalProviderUsed.has(provider)) {
```

Comment:
Afaict `generalProviderUsed` will always be empty and this lookup will always resolve to false?